omc3/optics_measurements/phase.py

```diff
@@ -48,35 +83,48 @@ def calculate(
         is a dictionary with the measured phase advances for 'free' and 'uncompensated' cases, as well as
         the location of the output ``TfsDataFrames`` for the phases.
     """
-    if meas_input.compensation == "none":
-        phase_advances, dfs = _calculate_with_compensation(meas_input,
-                                                           input_files,
-                                                           tunes,
-                                                           plane,
-                                                           meas_input.accelerator.model,
-                                                           'none',
-                                                           no_errors)
+    # Clean up compensation mode
+    if meas_input.compensation is None:
+        meas_input.compensation = CompensationMode.NONE
+    else:
+        meas_input.compensation = meas_input.compensation.lower()
+
+    LOGGER.info("-- Run on free model")
+    phase_advances, dfs = _calculate_with_compensation(
+        meas_input,
+        input_files,
+        tunes,
+        plane,
+        meas_input.accelerator.model,
+        meas_input.compensation,
+        no_errors,
+    )
+
+    if meas_input.compensation == CompensationMode.NONE:
         uncompensated_phase_advances = phase_advances
     else:
-        phase_advances, free_dfs = _calculate_with_compensation(meas_input,
-                                                                input_files,
-                                                                tunes,
-                                                                plane,
-                                                                meas_input.accelerator.model,
-                                                                meas_input.compensation,
-                                                                no_errors)
-        LOGGER.info("-- Run uncompensated")
-        uncompensated_phase_advances, drv_dfs = _calculate_with_compensation(meas_input,
-                                                                             input_files,
-                                                                             tunes,
-                                                                             plane,
-                                                                             meas_input.accelerator.model_driven,
-                                                                             'none',
-                                                                             no_errors)
-        dfs = free_dfs + drv_dfs
-
-
-    if len(phase_advances["MEAS"].index) < 3:
+        LOGGER.info("-- Run on driven model")
+
+        # TODO:Verify this is correct (that's how it was implemented before)
+        driven_compensation = (
+            CompensationMode.EQUATION 
+            if meas_input.compensation == CompensationMode.EQUATION else 
+            CompensationMode.NONE
+        )
```

Comment:
@lmalina @emaclean @awegsche @rogeliotomas 
I tried to make this a bit clearer how the logic is here, as this was hidden within the function. But I am not sure if this is correct as it is:
The way it had been implemented is, that the equation compensation also operates on the driven data, but the model compensation only on the free data, if I see that correctly.

Could you please confirm that this is correct? My knowledge of these compensations is very limited.

Reply:
After some discussion and feedback from lukas, we are both relatively sure this is a bug. I will try to device a test